scripts/scr_get_item_names/scr_get_item_names.gml

```diff
@@ -0,0 +1,806 @@
+#macro ITEM_NAME_NONE "(None)"
+#macro ITEM_NAME_ANY "(any)"
+
+/// @description This function returns an array with common empty or wildcard names as needed.
+/// @param {bool} _with_none - Whether to include ITEM_NAME_NONE in the list.
+/// @param {bool} _with_any - Whether to include ITEM_NAME_ANY in the list.
+function get_none_or_any_item_names(_with_none=false, _with_any=false) {
+    var _item_names = [];
+    if (_with_none) {
+        array_push(_item_names, ITEM_NAME_NONE);
+    }
+    if (_with_any) {
+        array_push(_item_names, ITEM_NAME_ANY);
+    }
+    return _item_names;
+}
+
+/// @description This function returns the hard-coded list of ranged weapons.
+/// @param {array} _item_names - The list of ranged weapons to append to.
+/// @returns {array} The appended list.
+function push_marine_ranged_weapons_item_names(_item_names) {
+    var item_count = 23
+    var initial_size = array_length(_item_names);
+    array_resize(_item_names, initial_size + item_count);
+
+    var index = initial_size;
+    _item_names[index++] = "Archeotech Laspistol";
+    _item_names[index++] = "Assault Cannon";
+    _item_names[index++] = "Bolt Pistol";
+    _item_names[index++] = "Bolter";
+    _item_names[index++] = "Stalker Pattern Bolter";
+    _item_names[index++] = "Combiflamer";
+    _item_names[index++] = "Flamer";
+    _item_names[index++] = "Heavy Bolter";
+    _item_names[index++] = "Heavy Flamer";
+    _item_names[index++] = "Hellrifle";
+    _item_names[index++] = "Incinerator";
+    _item_names[index++] = "Integrated Bolter";
+    _item_names[index++] = "Lascannon";
+    _item_names[index++] = "Lascutter";
+    _item_names[index++] = "Meltagun";
+    _item_names[index++] = "Missile Launcher";
+    _item_names[index++] = "Multi-Melta";
+    _item_names[index++] = "Autocannon";
+    _item_names[index++] = "Plasma Gun";
+    _item_names[index++] = "Plasma Pistol";
+    _item_names[index++] = "Sniper Rifle";
+    _item_names[index++] = "Storm Bolter";
+    _item_names[index++] = "Webber"; // 23
+    // future: not needed when we stop using GML's deprecated Copy-on-Write behavior
+    return _item_names;
+}
+
+/// @description This function returns the hard-coded list of melee weapons.
+/// @param {array} _item_names - The list to append to.
+/// @returns {array} The appended list.
+function push_marine_melee_weapons_item_names(_item_names) {
+    var item_count = 15;
+    var initial_size = array_length(_item_names);
+    array_resize(_item_names, initial_size + item_count);
+
+    var index = initial_size;
```

Comment:
These functions get called in Draw right now, I believe they are called every frame when the equip menu is open. I was thinking one resize at the the start would be cheaper than each time. I was also thinking that pushes were better than concat since there would be a left over array to garbage collect each frame. However, I'm not sure how the compiler works if it optimizes such things.

Reply:
> These functions get called in Draw right now, I believe they are called every frame when the equip menu is open. I was thinking one resize at the the start would be cheaper than each time. I was also thinking that pushes were better than concat since there would be a left over array to garbage collect each frame. However, I'm not sure how the compiler works if it optimizes such things.

Yeah, this may be true. The fact that this runs every frame is dumb, but oh well. Out of the scope of this PR.
Using a single array, for loop and `array_push` is just more readable, imo.
I'll check this through a profiler. Maybe it's okay to leave it like this while it runs every frame.

Reply:
> These functions get called in Draw right now, I believe they are called every frame when the equip menu is open.

Can you please double-check. I think they are called one time, when you click your mouse and open the menu. Then the return is stored in a variable and refreshed on other clicks.

Reply:
I confirmed that they are not spammed as I thought, thankfully.

Looking at cf8aefae, I touch all the call locations. 5 are in Draw_0.gml files, 1 in Step_0.gml and 1 is indirectly in draw via `scr_ui_settings`; that leaves 2 calls in a Mouse_50.gml. I did not look closer than noticing that these were called in draw events.

Reply:
> I confirmed that they are not spammed as I thought, thankfully.

Then can you use a single array with item names and either a `for` loop with `array_push` or `array_foreach`, instead of these shenanigans?

Reply:
Or `array_concat`, if you're not bothered by the fact that it returns a separate array and not modifies the original.

Reply:
In further testing, the calls in obj_mass_equip are per frame once a menu is opened.